packages/core-transaction-pool/src/processor.ts

```diff
@@ -176,11 +172,30 @@ export class Processor implements TransactionPool.IProcessor {
             );
 
             return false;
-        } else if (transaction.expiration > 0 && transaction.expiration <= lastHeight + 1) {
+        }
+
+        const lastHeight: number = app
+            .resolvePlugin<State.IStateService>("state")
+            .getStore()
+            .getLastHeight();
+
+        const maxTransactionAge: number = app.resolveOptions("transaction-pool").maxTransactionAge;
+
+        const expirationContext = {
+            blockTime: Managers.configManager.getMilestone(lastHeight).blocktime,
+            currentHeight: lastHeight,
+            now: Crypto.Slots.getTime(),
+            maxTransactionAge,
```

Comment:
Use `app.resolveOptions("transaction-pool").maxTransactionAge` directly here and ditch the `maxTransactionAge` variable - it is used in just one place.